lib/import/colorectal/providers/sheffield/sheffield_handler_colorectal.rb

```diff
@@ -7,618 +7,413 @@ module Providers
+            end
+            genocolorectals
+          end
+
+          def full_screen?(genocolorectal)
+            return if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Full screen/i).size.positive?
+          end
+
+          def targeted?(genocolorectal)
+            return if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Targeted/i).size.positive?
+          end
+
+          def no_scope?(genocolorectal)
+            return if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Unable/i).size.positive?
+          end
+
+          def process_fullscreen_records(genocolorectal, record, genocolorectals)
+            genotype_str = record.raw_fields['genotype']
+            if normal?(genotype_str)
+              process_normal_full_screen(genocolorectal, genocolorectals)
+            elsif positive_cdna?(genotype_str) || positive_exonvariant?(genotype_str)
+              process_variant_fs_records(genocolorectal, record, genocolorectals)
+            elsif only_protein_impact?(genotype_str)
+              process_only_protein_rec(genocolorectal, record, genocolorectals)
+              positive_gene = get_gene(record)
+              negative_genes = @genes_set - positive_gene
+              add_other_genes_with_status(negative_genes, genocolorectal, genocolorectals, 1)
+            elsif genotype_str.scan(/see\sbelow|comments/ix).size.positive?
+              add_other_genes_with_status(@genes_set, genocolorectal, genocolorectals, 4)
+            end
+          end
+
+          def process_targeted_no_scope_records(genocolorectal, record, genocolorectals)
+            genotype_str = record.raw_fields['genotype']
+            if normal?(genotype_str)
+              process_normal_targeted(genocolorectal, record, genocolorectals)
+            elsif positive_cdna?(genotype_str) || positive_exonvariant?(genotype_str)
+              process_variant_targeted(genocolorectal, record, genocolorectals)
+            elsif only_protein_impact?(genotype_str)
+              process_only_protein_rec(genocolorectal, record, genocolorectals)
+            end
+          end
+
+          def normal?(genotype_str)
+            genotype_str.scan(NORMAL_VAR_REGEX).size.positive?
+          end
+
+          def process_normal_targeted(genocolorectal, record, genocolorectals)
+            gene = get_gene(record)
+            gene = @genes_set&.size == 1 && gene.empty? ? @genes_set[0] : gene[0]
+            genocolorectal.add_gene_colorectal(gene)
+            genocolorectal.add_status(1)
+            genocolorectals.append(genocolorectal)
+            genocolorectals
+          end
+
+          def process_normal_full_screen(genocolorectal, genocolorectals)
+            negative_genes = @genes_set
```

Comment:
I have assigned it to negative_genes variable for making understandability of code clear as in case of normal full screen cases all the genes from the set have to be recorded as negative tests.